src/manage_nsfs/manage_nsfs_glacier.js

```diff
@@ -36,13 +36,38 @@ async function process_migrations() {
  */
 async function run_glacier_migrations(fs_context, backend) {
     // This WAL is getting opened only so that we can process all the prcess WAL entries
-    const wal = new PersistentLogger(
+    const wal = new PersistentLogger(config.NSFS_GLACIER_LOGS_DIR, GlacierBackend.MIGRATE_WAL_NAME, { locking: 'EXCLUSIVE' });
+    const failure_wal = new PersistentLogger(
         config.NSFS_GLACIER_LOGS_DIR,
-        GlacierBackend.MIGRATE_WAL_NAME,
-        { disable_rotate: true, locking: 'EXCLUSIVE' },
+        `${GlacierBackend.MIGRATE_WAL_NAME}.failure`,
+        { locking: 'EXCLUSIVE' },
     );
 
-    await wal.process_inactive(async file => backend.migrate(fs_context, file));
+    try {
+        // Process all the inactive and currently active log
+        await wal.process_inactive(async file => backend.migrate(fs_context, file, failure_wal.append.bind(failure_wal)));
+    } catch (error) {
+        console.error('failed to process restore logs:', error);
+    } finally {
+        await wal.close();
```

Comment:
what about closing failure_wal? perhaps we need to wrap this entire flow with a finally that will handle closing the both of them.